### ISSUE TITLE
Presentation crashes on api route due to resolvers (Main Document)

### ISSUE THREAD

**Issue by mdillgen** [sanity-io/visual-editing#1685]
I encountered a weird “bug”/behaviour regarding the Presentation Resolver API since Sanity v3.47.
I have a Next.js Setup with embedded studio based on the great Personal Website Template.  It's worth mentioning that I implemented i18n routing on it. The Presentation tool crashes when entering via the Presentation Tab in the upper center of the Studio UI.  

**To Reproduce**
1. Click ‘Presentation’ in the studio UI.
2. The internal browser then enters the api/draft route 
3. Studio crashes (seems it tries to resolve a main document for it)

```
Error: Cannot read properties of null (reading 'title')
TypeError: Cannot read properties of null (reading 'title')
    at getMissingDocumentFallback (webpack-internal:///(app-pages-browser)/./node_modules/sanity/lib/index.mjs:29209:104)
    at getPreviewValueWithFallback (webpack-internal:///(app-pages-browser)/./node_modules/sanity/lib/index.mjs:29218:164)
    at eval (webpack-internal:///(app-pages-browser)/./node_modules/@sanity/presentation/dist/_chunks-es/PresentationTool.js:47:3896)
    at updateMemo (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom-experimental/cjs/react-dom.development.js:12803:19)
    at Object.useMemo (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom-experimental/cjs/react-dom.development.js:13786:16)
    at useMemo (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-experimental/cjs/react.development.js:2588:21)
    at ln (webpack-internal:///(app-pages-browser)/./node_modules/@sanity/presentation/dist/_chunks-es/PresentationTool.js:47:3705)
    at renderWithHooks (webpack-internal:///(app-pages-browser)/./node_modules/next/dist/compiled/react-dom-experimental/cjs/react-dom.development.js:11182:18)
…

``` 

_Presentation setup in sanity.config.tsx:_
```
presentationTool({
      resolve,
      previewUrl: {
        previewMode: {
          enable: '/api/draft',
        },
      },
    }),
```
Main Document definitions in resolve.ts:_
```
export const mainDocuments = defineDocuments([
  {
    route: '/:language',
    filter: `_type == "home" && language == $language`,
  },
  {
    route: '/:language/events',
    filter: `_type == "events" && language == $language`,
  },
  …
  
```
Working example (resolve anything for the api route):
```
…
  {
    route: '/api/draft',
    filter: `_type == "home" && language == "de"`,
  },
…
```

**Which versions of Sanity are you using?**
```
@sanity/cli (global)                   3.46.1 (latest: 3.47.1)
@sanity/document-internationalization   3.0.0 (up to date)
@sanity/image-url                       1.0.2 (up to date)
@sanity/preview-url-secret             1.6.17 (up to date)
@sanity/react-loader                   1.10.3 (up to date)
@sanity/types                          3.47.1 (up to date)
@sanity/ui                              2.4.0 (up to date)
@sanity/vision                         3.47.1 (up to date)
sanity                                 3.47.1 (up to date)
```

**What operating system are you using?**

**Which versions of Node.js / npm are you running?**
```
npm 10.1.0
Node v20.9.0
```

**Comment by mariusGundersen** [comment]
I'm having the same problem. It crashes when it goes to a route without a document, since it tries to show a "missing document" component without being able to get the document title.
![bilde](https://github.com/sanity-io/visual-editing/assets/464152/4bcbc343-9ccd-4e26-be6c-9315f8beb204)
As you can see here the `value` is `null`, so this fails. This method is called from [ContentEditor](https://github.com/sanity-io/visual-editing/blob/1f230a3b68bd55713997a11937acdc9508574b68/packages/presentation/src/editor/ContentEditor.tsx#L78), where it tries to get `mainDocumentState.document` and it forces itself past TypeScript which warns that it might not be st
![bilde](https://github.com/sanity-io/visual-editing/assets/464152/09f5890e-5072-4271-861b-f6c57aab7708)
The solution is probably to check `mainDocumentState?.document` on this line.

**Comment by mdillgen** [comment]
@mariusGundersen Thanks for the clarification! Hopefully it will be fixed.

**Comment by mariusGundersen** [comment]
I tested the fix and with that one change it now produces this:
![bilde](https://github.com/sanity-io/visual-editing/assets/464152/0ffe5354-9da4-4bde-9f9f-86b7ac037c24)

**Comment by rdunk** [comment]
Thanks for this, the fix should be included in [v1.16.1](https://github.com/sanity-io/visual-editing/releases/tag/presentation-v1.16.1).
